internal/bminventory/inventory.go

```diff
@@ -600,6 +604,19 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	if b.ocmClient != nil {
```

Comment:
In what scenario the client is nil?

Reply:
In case ENABLE_AUTH=false

Reply:
Then please create a dummy client instead.

Reply:
besides, i didn't see any nil configuration in the main so  where do you set it to nil?

Reply:
This is existing code (not part of the PR), it is declared [here](https://github.com/ybettan/assisted-service/blob/7fa60b0f3839f2227412bcdf1afb37d4f2fd7405/cmd/main.go#L144) and in case `ENABLE_AUTH` is turn off then it is not set to any real client [here](https://github.com/ybettan/assisted-service/blob/7fa60b0f3839f2227412bcdf1afb37d4f2fd7405/cmd/main.go#L147) what means it stays the "0" value of a ptr which is nil.

Reply:
I think it will be clearer to use a dummy client instead of checking for nil, if someone else will add a code that try to use this client he may not check if it's nil or not.

Reply:
Would it be better if I use `if EnableAuth` instead of `if ocmClient != nil` ?
I am not sure what you mean by using a "dummy client"

Reply:
in both cases it will not be clear, how about adding all this logic inside cluster api?

Reply:
obsolete